functors_task_1/student/include/impl.h

```diff
@@ -40,9 +66,31 @@ struct MixingPolicy
      * Whiskey + SevenUp = SevenPlusSeven;
      * Others + Others = Oops;
      */
-    static Cocktail mix(/**???*/)
+    static Cocktail mix(AlcoholDrink alchohol, NonAlcoholDrink nonAlcohol)
+    {
+        std::unordered_map<AlcoholDrink, std::unordered_map<NonAlcoholDrink, Cocktail>> cocktailes =
+        {
+            { AlcoholDrink::Gin,     { {NonAlcoholDrink::LimeJuice, Cocktail::Gimlet},
+                                       {NonAlcoholDrink::GrapefruitJuice, Cocktail::Greyhount} } },
+
+            { AlcoholDrink::Whiskey, { {NonAlcoholDrink::SevenUp, Cocktail::SevenPlusSeven} } }
+        };
+
+        if( isFound(cocktailes, alchohol) &&
+                isFound(cocktailes[alchohol], nonAlcohol) )
```

Comment:
if you already checked that the map contains value, why do you need to use "[]" operator?